src/bc.F

```diff
@@ -26,23 +19,25 @@ MODULE bc_module
 !ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 
 
-      subroutine bcp(p)
+      subroutine bcp(p,device)
       use input, only : ib,ie,jb,je,kb,ke,wbc,ebc,sbc,nbc, &
           ibw,ibe,ibs,ibn,ni,nj,nk,timestats,time_bc,mytime
       implicit none
 
       real, dimension(ib:ie,jb:je,kb:ke) :: p
+      logical, optional, intent(in) :: device
 
       integer i,j,k
-      !$acc declare present(p)
+      logical :: openacc 
 
+      call SetGPUParams(openacc,device)
```

Comment:
Maybe we need to add `!$acc declare copyin(XXX)` for all the **bcXX** subroutines here?

Reply:
I don't quite understand this comment.  Maybe we need to discuss on a video call.

Reply:
After a discussion with @johnmauff , we do not need the `!$acc declare` statement in the **bcXX** subroutines and all the data copy should happen in the `cm1.F` code.

The `default (present)` clause checks that all the variables for the **bcXX** subroutines should be GPU-resident and an error message will be issued if it is not the case.